### DISCUSSION TITLE
Heh, finally a CLI framework I could use

### DISCUSSION THREAD

**Discussion by pawamoy** [treykeown/arguably#20] [General]
After 8 years writing Python, I'm still using argparse. Click never convinced me, nor did Typer (or Fire). They ask me to learn how they work, they change the objects I decorate with, they make it hard or impossible to call the original functions programmatically (in Python, not from the CLI). They force system exits. They are not flexible enough for some of my use-cases.

argparse is super simple and flexible but makes my code quite verbose, I admit.

arguably seems like it fixes most of these pain points :slightly_smiling_face: I'll be sure to give it a try! Great job, and thanks for sharing with the community! :rocket:

**Comment by treykeown** [comment]
Hey, thanks for the kind words! It's been awesome seeing people using what started as a couple functions for my scripts. `arguably` is still fresh and might have some rough edges. Would love to hear more on your experience with it when you do give it a try!

**Comment by BruceEckel** [comment]
I like what I've seen so far but development is looking pretty quiet lately -- is this project being developed?

I notice it's not mentioned here: https://github.com/shadawck/awesome-cli-frameworks?tab=readme-ov-file#python

**Reply by treykeown** [reply]
Hi! I'm definitely still keeping this going, just haven't had time recently to write code here. After the holidays I'll be pushing some updates.

I didn't know that list existed, I'll definitely open a PR.
